### PR TITLE
Refactor validaton package

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Comment by 6543** [comment]
->   #19479

File: modules/git/repo_branch.go [hunk 1]
```diff
@@ -151,6 +154,13 @@ func (repo *Repository) CreateBranch(branch, oldbranchOrCommit string) error {
 
 // AddRemote adds a new remote to repository.
 func (repo *Repository) AddRemote(name, url string, fetch bool) error {
+	if db.AlphaDashDotPattern.MatchString(name) {
```
Comment (lunny) [inline modules/git/repo_branch.go]: git module should not reference db module

File: modules/validation/address.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"net/url"
+	"strings"
+)
+
+// ErrInvalidCloneAddr tell us a remote address is invalid
+var ErrInvalidCloneAddr = errors.New("remote address is invalid")
+
+// ParseRemoteAddr checks if given remote address is valid,
+// and returns composed URL with needed username and password.
+func ParseRemoteAddr(remoteAddr, authUsername, authPassword string) (string, error) {
```
Comment (lunny) [inline modules/validation/address.go:18]: I have a similar change in https://github.com/go-gitea/gitea/pull/18649/files

**Comment by lunny** [comment]
Is this still WIP?

**Comment by 6543** [comment]
No time until next thusday to work on anything ...

**Comment by wxiaoguang** [comment]
Stale for long time, is it still in progress?

IIRC this is for some security problems, while these security problems have been fixed?

**Comment by wxiaoguang** [comment]
Kindly second ping ~~~

**Comment by wxiaoguang** [comment]
Maybe this PR could be closed? Feel free to complete the description and reopen.

**Pull request closed without merging**